File: app/src/main/java/com/tj/skyone/ui/login/LoginActivity.java
package com.tj.skyone.ui.login;

import android.text.Editable;
import android.text.TextWatcher;
import android.widget.EditText;
import android.widget.TextView;

import com.blankj.utilcode.util.GsonUtils;
import com.blankj.utilcode.util.JsonUtils;
import com.blankj.utilcode.util.SPUtils;
import com.blankj.utilcode.util.StringUtils;
import com.blankj.utilcode.util.ToastUtils;
import com.google.gson.Gson;
import com.tj.skyone.R;
import com.tj.skyone.base.BaseActivity;
import com.tj.skyone.ui.home.view.HomeActivity;
import com.tj.skyone.utils.NoDoubleClickUtils;
import com.tj.skyone.utils.TcpClient;
import com.tj.skyone.utils.eventbus.AnyEventType;
import com.tj.skyone.utils.eventbus.AnyEventTypes;
import com.tj.skyone.utils.eventbus.EventBusUtils;

import org.greenrobot.eventbus.Subscribe;
import org.json.JSONObject;

import java.util.HashMap;

import butterknife.BindView;
import butterknife.OnClick;

public class LoginActivity extends BaseActivity {


    @BindView(R.id.et_phone)
    EditText etPhone;
    @BindView(R.id.et_pwd_or_code)
    EditText etPwdOrCode;
    @BindView(R.id.btn_login)
    TextView btnLogin;

    @Override
    protected int getLayoutId() {
        return R.layout.activity_login;
    }

    @Override
    protected void initEventAndData() {

        EventBusUtils.register(this);

        etPhone.addTextChangedListener(new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
            }

            @Override
            public void onTextChanged(CharSequence s, int start, int before, int count) {
            }

            @Override
            public void afterTextChanged(Editable s) {


                if (!StringUtils.isEmpty(etPhone.getText())&& !StringUtils.isEmpty(etPwdOrCode.getText())){


                    btnLogin.setBackground(getResources().getDrawable(R.mipmap.loginbs));

                }else{

                    btnLogin.setBackground(getResources().getDrawable(R.mipmap.loginb));

                }

            }
        });

        etPwdOrCode.addTextChangedListener(new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
            }

            @Override
            public void onTextChanged(CharSequence s, int start, int before, int count) {
            }

            @Override
            public void afterTextChanged(Editable s) {

                if (!StringUtils.isEmpty(etPhone.getText())&& !StringUtils.isEmpty(etPwdOrCode.getText())){


                    btnLogin.setBackground(getResources().getDrawable(R.mipmap.loginbs));

                }else{

                    btnLogin.setBackground(getResources().getDrawable(R.mipmap.loginb));

                }

            }

        });


    }

    @Override
    protected void isStart() {

    }

    @Override
    public void showError(String errorMessage) {

    }

    @Override
    public void showResult(Object object) {

    }

    @Override
    public void showResult(Object object, String code) {

    }


    @OnClick(R.id.btn_login)
    public void onViewClicked() {

        if (!NoDoubleClickUtils.isDoubleClick()) {


            if (StringUtils.isEmpty(etPhone.getText())){
                ToastUtils.showLong("请输入用户名");
                return;
            }

            if (StringUtils.isEmpty(etPwdOrCode.getText())){
                ToastUtils.showLong("请输入密码");
                return;
            }


            HashMap map = new HashMap();

            map.put("methodName","appLogin");
            map.put("loginName",etPhone.getText().toString());
            map.put("loginPassword",etPwdOrCode.getText().toString());


            TcpClient.getInstance().sendChsPrtCmds(new Gson().toJson(map), 1001);

            getDialog().show();



        }

    }

    @Subscribe
    public void onEvent(AnyEventTypes event) {

        getDialog().dismiss();

        if (StringUtils.equals("appLogin",event.getEventCode())) {



            SPUtils.getInstance().put("user", etPhone.getText().toString());
            readyGo(HomeActivity.class);
            finish();

        }



    }

    @Override
    protected void onDestroy() {
        EventBusUtils.unregister(this);
        super.onDestroy();
    }
}

File: app/src/main/java/com/tj/skyone/ui/login/LoginActivity.kt
package com.tj.skyone.ui.login

import android.content.Intent
import android.content.pm.ActivityInfo
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.core.content.ContextCompat
import androidx.core.widget.addTextChangedListener
import com.blankj.utilcode.util.SPUtils
import com.blankj.utilcode.util.ServiceUtils
import com.blankj.utilcode.util.StringUtils
import com.blankj.utilcode.util.ToastUtils
import com.google.gson.Gson
import com.tj.skyone.R
import com.tj.skyone.databinding.ActivityLoginBinding
import com.tj.skyone.service.MyService
import com.tj.skyone.ui.GlobalApp.Companion.isPad
import com.tj.skyone.ui.home.view.HomeActivity
import com.tj.skyone.utils.TcpClient
import com.tj.skyone.utils.eventbus.AnyEventTypes
import com.tj.skyone.utils.eventbus.EventBusUtils
import com.tj.skyone.widget.dialog.LoadingDialog
import me.jessyan.autosize.internal.CustomAdapt
import org.greenrobot.eventbus.Subscribe

class LoginActivity : AppCompatActivity(), CustomAdapt {
    private lateinit var binding: ActivityLoginBinding
    private lateinit var dialog: LoadingDialog

    private val tag = "LoginActivity"

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        requestedOrientation = if(isPad) {
            //锁横屏
            ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE
        } else {
            ActivityInfo.SCREEN_ORIENTATION_PORTRAIT
        }

        binding = ActivityLoginBinding.inflate(layoutInflater)
        setContentView(binding.root)
        initEventData()
    }

    private fun initEventData() {
        //开启后台服务
        try {
            if (!ServiceUtils.isServiceRunning(MyService::class.java))
                ServiceUtils.startService(MyService::class.java)
        } catch (e: Exception) {
            e.printStackTrace()
        }
        //注册EventBus
        EventBusUtils.register(this)
        //初始化加载框
        dialog = LoadingDialog(this)

        //设置密码输入框监听时间
        binding.etPwdOrCode.addTextChangedListener {
            val content = it.toString()
            if (content.isNotEmpty()) {
                binding.btnLogin.background = ContextCompat.getDrawable(this, R.mipmap.loginbs)
            }
        }

        //设置登录按钮监听
        binding.btnLogin.setOnClickListener {
            when {
                StringUtils.isEmpty(binding.etPhone.text) -> {
                    ToastUtils.showLong("请输入用户名")
                }
                StringUtils.isEmpty(binding.etPwdOrCode.text) -> {
                    ToastUtils.showLong("请输入密码")
                }
                else -> {
                    val map = HashMap<Any, Any>()
                    map["methodName"] = "appLogin"
                    map["loginName"] = binding.etPhone.text.toString()
                    map["loginPassword"] = binding.etPwdOrCode.text.toString()
                    TcpClient.getInstance().sendChsPrtCmds(Gson().toJson(map), 1001)
                    dialog.show()
                }
            }
        }
    }

    @Subscribe
    fun onEvent(event: AnyEventTypes) {
        dialog.dismiss()
        if (StringUtils.equals("appLogin", event.eventCode)) {
            SPUtils.getInstance().put("user", binding.etPhone.text.toString())
            val intent = Intent(this, HomeActivity::class.java)
            startActivity(intent)
            finish()
        }
    }

    override fun isBaseOnWidth(): Boolean {
        return true
    }

    override fun getSizeInDp(): Float {
        return if (isPad) {
            853.0.toFloat()
        } else {
            392.0.toFloat()
        }
    }
}


